flink-core/src/main/java/org/apache/flink/core/plugin/PluginManagerImpl.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.core.plugin;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.annotation.VisibleForTesting;
+
+import org.apache.flink.shaded.guava18.com.google.common.collect.Iterators;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+
+/**
+ * Implementation of {@link PluginManager}.
+ */
+@Internal
+@ThreadSafe
+public class PluginManagerImpl implements PluginManager {
```

Comment:
I would suggest to pick another name because there might be multiple `PluingManager` implementations. Maybe one could name this implementation `DefaultPluginManager` or `BasicPluginManager`. Admittedly, this name is not a lot better but I think it is a bit better than `Impl`.

Reply:
I think `DefaultPluginManager` might be good enough. Sorry that I'm not good at naming or documenting.